libautoscoper/src/KeyCurve.hpp

```diff
@@ -254,4 +267,198 @@ class KeyCurve
     key_map keys;
 };
 
+// Need to have the implementation here in the header because of the template
+
+template <typename T>
+void KeyCurve<T>::insert(int time)
+{
+  insert(time, (*this)(time));
+}
+template <typename T>
+void KeyCurve<T>::insert(int time, T value)
+{
+  Key *key = new Key;
+  key->value = value;
+  key->in_tangent_type = SMOOTH;
+  key->out_tangent_type = SMOOTH;
+  key->bind_tangents = true;
+  key->in_tangent_lock = false;
+  key->out_tangent_lock = false;
+
+  iterator position = keys.find(time);
+  if (position == keys.end()) {
+    std::pair<iterator, bool> temp = keys.insert(std::make_pair(time, key));
+    position = temp.first;
```

Comment:
The flickering radius issue is happening consistently across these frames [69, 108, 156, 179, 219, 224, 233] and only for the radius in the WN00113 dataset (at least as far as I can tell)

From what I can tell is happening (using frame 69 as the example), when we go to insert frame 70 into the map some sort of memory leak or something is happening that is causing frame 69 to get overwritten.

Right before we insert frame 70:
```
>? this->operator()(69)
{w=0.0612879843 x=0.0195402186 y=0.693150759 ...}
    w: 0.0612879843
    x: 0.0195402186
    y: 0.693150759
    z: -0.717916369
    data: 0x000000c25ecf7d70 {0.0612879843, 0.0195402186, 0.693150759, -0.717916369}
```

Right after we insert frame 70:
```
>? this->operator()(69)
{w=-nan(ind) x=-nan(ind) y=-nan(ind) ...}
    w: -nan(ind)
    x: -nan(ind)
    y: -nan(ind)
    z: -nan(ind)
    data: 0x000000c25ecf7d70 {-nan(ind), -nan(ind), -nan(ind), -nan(ind)}
>
```